server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -859,60 +875,71 @@ public TSFetchResultsResp fetchResults(TSFetchResultsReq req) {
 
       QueryDataSet queryDataSet;
       if (!queryId2DataSet.get().containsKey(req.queryId)) {
-        queryDataSet = createNewDataSet(req);
+        queryDataSet = createNewDataSet(req.queryId);
```

Comment:
Users may call  "executeQueryStatement(TSExecuteStatementReq req)"  in line 651 to execute a query. The dataset needs to be also added in this function. 

It's better to move the fill data into executeQueryStatement(long statementId, PhysicalPlan plan), then the if statement in fetchResults could be removed, becuase the dataset must be saved in queryId2DataSet and queryId2Plan.